File: tests/Generator/resource/swift/oop/Map.swift
class Map<P, T>: Codable {
    var totalResults: Int
    var parent: P
    var entries: Array<T>

    enum CodingKeys: String, CodingKey {
        case totalResults = "totalResults"
        case parent = "parent"
        case entries = "entries"
    }
}

File: tests/Generator/resource/rust/oop/map.rs
use serde::{Serialize, Deserialize};

#[derive(Serialize, Deserialize)]
pub struct Map {
    #[serde(rename = "totalResults")]
    total_results: Option<u64>,

    #[serde(rename = "parent")]
    parent: Option<P>,

    #[serde(rename = "entries")]
    entries: Option<Vec<T>>,

}

